### DISCUSSION TITLE
Let's make our SPA more secure by setting up a .NET BFF with Duende and Auth0

### DISCUSSION THREAD

**Discussion by giscus** [timdeschryver/timdeschryver.dev#299] [Announcements]
# Let's make our SPA more secure by setting up a .NET BFF with Duende and Auth0

To improve the security of our Single Page Applications, let's ditch the access tokens in the browser and make the Backend for Frontend (BFF) responsible for the authentication process using the Authorization Code flow with PKCE. To achieve this, we'll use the Duende.BFF NuGet package and let it communicate with Auth0.

https://timdeschryver.dev/blog/lets-make-our-spa-more-secure-by-setting-up-a-net-bff-with-duende-and-auth0

**Comment by akhanalcs** [comment]
Can you please create another blog post where you consume this API from the SPA side? For eg: Angular 17.
Thanks!

**Reply by timdeschryver** [reply]
See https://timdeschryver.dev/blog/lets-make-our-spa-more-secure-by-consuming-a-duende-bff-with-angular

(There's a link at the bottom of the post, but I guess I should make this "pop-out" more)

**Comment by lorinczantonia** [comment]
Great article! I was wondering if Azure can ve used instead of Auth0?

**Reply by lorinczantonia** [reply]
Be*

**Reply by timdeschryver** [reply]
@lorinczantonia Azure can also be used.
